libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -175,28 +175,28 @@ void CodeTransform::operator()(VariableDeclaration const& _varDecl)
 {
 	yulAssert(m_scope, "");
 
-	int const numVariables = _varDecl.variables.size();
+	size_t const numVariables = _varDecl.variables.size();
 	int heightAtStart = m_assembly.stackHeight();
 	if (_varDecl.value)
 	{
 		std::visit(*this, *_varDecl.value);
-		expectDeposit(numVariables, heightAtStart);
+		expectDeposit(int(numVariables), heightAtStart);
 	}
 	else
 	{
 		m_assembly.setSourceLocation(_varDecl.location);
-		int variablesLeft = numVariables;
+		size_t variablesLeft = numVariables;
 		while (variablesLeft--)
 			m_assembly.appendConstant(u256(0));
 	}
 
 	m_assembly.setSourceLocation(_varDecl.location);
 	bool atTopOfStack = true;
-	for (int varIndex = numVariables - 1; varIndex >= 0; --varIndex)
+	for (size_t varIndex = numVariables - 1; varIndex < numeric_limits<size_t>::max(); --varIndex)
```

Comment:
This should work but looks really unnatural. I think it would be better for `varIndex` to be `int` and to cast to `size_t` as needed instead.

Reply:
I was thinking the same, it is really a bit tricky, so will do like suggested